<?xml version="1.0"?>
<launch>
  <arg name="namespace" default="flir_boson"/>
  <arg name="frame_id" default="boson_camera"/>

  <!-- the linux file descriptor location for the camera -->
  <arg name="left_ip" default="169.254.87.157"/>
  <!-- <arg name="right_ip" default="169.254.94.215"/> -->
  <arg name="right_ip" default="169.254.82.142"/>

  <arg name="left_namespace" default="left" />
  <arg name="right_namespace" default="right" />
    
  <!-- valid values are 30.0 or 60.0 for Bosons -->
  <arg name="frame_rate" default="60.0"/>

  <!-- valid values are RAW16 or YUV -->
  <arg name="video_mode" default="YUV"/>

  <!-- valid values are TRUE or FALSE -->
  <arg name="zoom_enable" default="FALSE"/>

  <!-- valid values are Boson_320 or Boson_640 -->
  <!-- <arg name="sensor_type" default="Boson_640"/> -->

  <!-- location of the camera calibration file -->
  <arg name="camera_info_url" default="package://flir_boson_ethernet/example_calibrations/BlackFlyS.yaml"/>

  <node pkg="nodelet" 
        type="nodelet"
        name="nodelet_manager"
        ns="$(arg namespace)"
        args="manager"
        output="screen"/>

  <node pkg="nodelet" 
        type="nodelet"
        name="left_camera_node"
        args="load flir_boson_ethernet/CameraController nodelet_manager"
        ns="$(arg namespace)"
        output="screen">
    <remap from="image_raw" to="$(arg left_namespace)/image_raw" />
    <remap from="camera_info_url" to="$(arg left_namespace)/camera_info_url" />
    <remap from="set_camera_info" to="$(arg left_namespace)/set_camera_info" />
    
    <param name="frame_id" type="str" value="$(arg frame_id)"/>
    <param name="ip_addr" type="str" value="$(arg left_ip)"/>
    <param name="frame_rate" type="double" value="$(arg frame_rate)"/>
    <param name="video_mode" type="str" value="$(arg video_mode)"/>
    <param name="zoom_enable" type="bool" value="$(arg zoom_enable)"/>
    <param name="camera_info_url" type="str" value="$(arg camera_info_url)"/>
  </node>

  <node pkg="nodelet" 
        type="nodelet"
        name="right_camera_node"
        args="load flir_boson_ethernet/CameraController nodelet_manager"
        ns="$(arg namespace)"
        output="screen">
    <remap from="image_raw" to="$(arg right_namespace)/image_raw" />
    <remap from="camera_info_url" to="$(arg right_namespace)/camera_info_url" />
    <remap from="set_camera_info" to="$(arg right_namespace)/set_camera_info" />

    <param name="frame_id" type="str" value="$(arg frame_id)"/>
    <param name="ip_addr" type="str" value="$(arg right_ip)"/>
    <param name="frame_rate" type="double" value="$(arg frame_rate)"/>
    <param name="video_mode" type="str" value="$(arg video_mode)"/>
    <param name="zoom_enable" type="bool" value="$(arg zoom_enable)"/>
    <param name="camera_info_url" type="str" value="$(arg camera_info_url)"/>
  </node>

  <node name="left_view"
        pkg="image_view"
        type="image_view"
        output="screen">
    <remap from="image" to="flir_boson/left/image_raw" />
  </node>
  <node name="right_view"
        pkg="image_view"
        type="image_view"
        output="screen">
    <remap from="image" to="flir_boson/right/image_raw" />
  </node>

</launch>
